test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeGenerator.java

```diff
@@ -198,158 +181,157 @@ private static void testWithFieldOnNonExistentClass(ClassWriter cw, String class
 			case "J":
+				mv.visitVarInsn(LLOAD, count);
 				doubleDetected = true;
-				mv.visitVarInsn(LLOAD, i);
+				count += 2;
 				break;
 			default:
-				mv.visitVarInsn(ALOAD, i);
+				mv.visitVarInsn(ALOAD, count);
+				count++;
 				break;
 			}
 			mv.visitFieldInsn(WITHFIELD, valueName, nameAndSig[0], nameAndSig[1]);
 		}
 		mv.visitInsn(ARETURN);
-		
 		int maxStackAndLocals = (doubleDetected) ? 3 : 2;
-		mv.visitMaxs(maxStackAndLocals, fields.length);
+		mv.visitMaxs(makeMaxLocal, makeMaxLocal);
 		mv.visitEnd();
 	}
 	
 	private static void makeGeneric(ClassWriter cw, String className, String methodName, String specificMethodName, String makeValueSig, String makeValueGenericSig, String[] fields, int makeMaxLocal) {
-		int maxStack = 0;
+		boolean doubleDetected = false;
 		MethodVisitor mv = cw.visitMethod(ACC_PUBLIC  + ACC_STATIC, methodName, "(" + makeValueGenericSig + ")L" + className + ";", null, new String[] {"java/lang/Exception"});
 		mv.visitCode();
 		for (int i = 0; i <  fields.length; i++) {
 			mv.visitVarInsn(ALOAD, i);
 			String nameAndSigValue[] = fields[i].split(":");
 			switch (nameAndSigValue[1]) {
 			case "D":
-				maxStack += 2;
 				mv.visitTypeInsn(CHECKCAST, "java/lang/Double");
 				mv.visitMethodInsn(INVOKEVIRTUAL, "java/lang/Double", "doubleValue", "()D", false);
+				doubleDetected = true;
 				break;
 			case "I":
-				maxStack++;
 				mv.visitTypeInsn(CHECKCAST, "java/lang/Integer");
 				mv.visitMethodInsn(INVOKEVIRTUAL, "java/lang/Integer", "intValue", "()I", false);
 			case "Z":
-				maxStack++;
 				mv.visitTypeInsn(CHECKCAST, "java/lang/Boolean");
 				mv.visitMethodInsn(INVOKEVIRTUAL, "java/lang/Boolean", "booleanValue", "()Z", false);
 			case "B":
-				maxStack++;
 				mv.visitTypeInsn(CHECKCAST, "java/lang/Byte");
 				mv.visitMethodInsn(INVOKEVIRTUAL, "java/lang/Byte", "byteValue", "()B", false);
 			case "C":
-				maxStack++;
 				mv.visitTypeInsn(CHECKCAST, "java/lang/Character");
 				mv.visitMethodInsn(INVOKEVIRTUAL, "java/lang/Character", "charValue", "()C", false);
 			case "S":
-				maxStack++;
 				mv.visitTypeInsn(CHECKCAST, "java/lang/Short");
 				mv.visitMethodInsn(INVOKEVIRTUAL, "java/lang/Short", "shortValue", "()S", false);
 				break;
 			case "F":
-				maxStack++;
 				mv.visitTypeInsn(CHECKCAST, "java/lang/Float");
 				mv.visitMethodInsn(INVOKEVIRTUAL, "java/lang/Float", "floatValue", "()F", false);
 				break;
 			case "J":
-				maxStack += 2;
 				mv.visitTypeInsn(CHECKCAST, "java/lang/Double");
 				mv.visitMethodInsn(INVOKEVIRTUAL, "java/lang/Double", "doubleValue", "()J", false);
+				doubleDetected = true;
 				break;
 			default:
-				maxStack++;
 				break;
 			}
 		}
-		
 		mv.visitMethodInsn(INVOKESTATIC, className, specificMethodName, "(" + makeValueSig + ")L"+ className + ";", false);
 		mv.visitInsn(ARETURN);
-		mv.visitMaxs(maxStack, fields.length);
+
+		int maxStackAndLocals = (doubleDetected) ? 3 : 2;
+		mv.visitMaxs(maxStackAndLocals, fields.length);
```

Comment:
this one should `makeMaxLocal` for both params as this function will take all the args and push them on the stack and call the make*